ARCHITECTURE.md

```diff
@@ -0,0 +1,17 @@
+# ARCHITECTURE
+
+This repository holds the source code for the Xilem project and the Masonry project, including their sub-crates.
+
+- Xilem is a family of high-level GUI frameworks. Xilem apps are written with idiomatic Rust code, with little to no reliance on macros and DSLs.
+    - **`xilem_core`** includes the traits that define Xilem.
+    - **`xilem`** is the native framework, built on Masonry.
+    - **`xilem_web`** is the web framework, built on the DOM.
+- Masonry is a foundational framework for building high-level Rust GUI libraries.
+    - **`masonry_core`** includes the core traits, types and abstractions.
```

Comment:
I never really liked this "core traits, types and abstractions" terminology; it's a very vague description. What do you think about the "base GUI engine" from https://github.com/linebender/xilem/pull/1426?

This isn't blocking.